<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>939</width>
    <height>732</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>520</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>地址访问：</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="addressStreamFile">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>550</y>
      <width>117</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>地址流文件：</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="inputByHand">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>620</y>
      <width>117</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>手动输入</string>
    </property>
   </widget>
   <widget class="QCommandLinkButton" name="stepByStep">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>120</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>步进</string>
    </property>
   </widget>
   <widget class="QCommandLinkButton" name="run">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>120</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>执行到底</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>60</y>
      <width>981</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>10</y>
      <width>181</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:24pt; color:#0000ff;&quot;&gt;Cache模拟器&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>70</y>
      <width>20</width>
      <height>801</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>80</y>
      <width>101</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:18pt; color:#ff0000;&quot;&gt;执行控制&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>140</y>
      <width>235</width>
      <height>60</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QRadioButton" name="uniCacheSelected">
         <property name="text">
          <string>统一Cache的大小：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="sizeCache">
         <property name="currentText">
          <string>2K</string>
         </property>
         <item>
          <property name="text">
           <string>2K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>4K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>8K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>16K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>32K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>64K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>128K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>256K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>512K</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1M</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QRadioButton" name="idCacheSelected">
       <property name="text">
        <string>独立Cache：</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>210</y>
      <width>201</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>数据cache的大小：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="dCacheSize">
       <item>
        <property name="text">
         <string>1K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>4K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>8K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>16K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>32K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>64K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>128K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>256K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>512K</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>250</y>
      <width>200</width>
      <height>29</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>指令cache的大小：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="iCacheSize">
       <item>
        <property name="text">
         <string>1K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>4K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>8K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>16K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>32K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>64K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>128K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>256K</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>512K</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>290</y>
      <width>265</width>
      <height>221</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>块大小：</string>
         </property>
         <property name="buddy">
          <cstring>sizeBlock</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="sizeBlock">
         <item>
          <property name="text">
           <string>32B</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>64B</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>128B</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>256B</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>512B</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>相连度：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="sizeWay">
         <item>
          <property name="text">
           <string>直接映射</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2路</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>4路</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>8路</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>16路</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>32路</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>替换策略：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="replaceMethod">
         <item>
          <property name="text">
           <string>LRU</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>FIFO</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>RAND</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>预取策略：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="prefetchMethod">
           <item>
            <property name="text">
             <string>不预取</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>不命中预取</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>写策略：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="writeMethod">
           <item>
            <property name="text">
             <string>写回法</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>写直达法</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>写不命中的调块策略：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="blockFetchMethod">
           <item>
            <property name="text">
             <string>按写分配</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>不按写分配</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>280</width>
      <height>38</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_10">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:18pt; color:#ff0000;&quot;&gt;设置参数&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCommandLinkButton" name="reset">
       <property name="text">
        <string>复位</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QCommandLinkButton" name="lookUp">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>580</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>浏览</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="filePath">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>580</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>-510</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>模拟结果</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>320</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>其中：</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>413</x>
      <y>268</y>
      <width>457</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_11">
     <item>
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>总访问次数：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="totalAccessNum">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_16">
       <property name="text">
        <string>不命中次数：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="totalMissNum">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_19">
       <property name="text">
        <string>不命中率：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="totalMissRate">
       <property name="text">
        <string>0.0%</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>415</x>
      <y>340</y>
      <width>451</width>
      <height>141</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_12">
       <item>
        <widget class="QLabel" name="label_26">
         <property name="text">
          <string>读指令次数：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="readINum">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_25">
         <property name="text">
          <string>不命中次数：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="readIMissNum">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_22">
         <property name="text">
          <string>不命中率：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="readIMissRate">
         <property name="minimumSize">
          <size>
           <width>54</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>0.0%</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_14">
       <item>
        <widget class="QLabel" name="label_32">
         <property name="text">
          <string>读数据次数：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="readDNum">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_31">
         <property name="text">
          <string>不命中次数：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="readDMissNum">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_28">
         <property name="text">
          <string>不命中率：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="readDMissRate">
         <property name="minimumSize">
          <size>
           <width>54</width>
           <height>41</height>
          </size>
         </property>
         <property name="text">
          <string>0.0%</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_13">
       <item>
        <widget class="QLabel" name="label_38">
         <property name="text">
          <string>写数据次数：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="writeNum">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_37">
         <property name="text">
          <string>不命中次数：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="writeMissNum">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_34">
         <property name="text">
          <string>不命中率：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="writeMissRate">
         <property name="minimumSize">
          <size>
           <width>54</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>0.0%</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>650</y>
      <width>211</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_15">
     <item>
      <widget class="QComboBox" name="cmdKind">
       <item>
        <property name="text">
         <string>读数据</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>写数据</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>读指令</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="address"/>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>200</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:18pt; color:#ff0000;&quot;&gt;执行结果&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>run</zorder>
   <zorder>label_10</zorder>
   <zorder>addressStreamFile</zorder>
   <zorder>inputByHand</zorder>
   <zorder>stepByStep</zorder>
   <zorder>line</zorder>
   <zorder>label_11</zorder>
   <zorder>line_2</zorder>
   <zorder>label_12</zorder>
   <zorder>lookUp</zorder>
   <zorder>filePath</zorder>
   <zorder>label_13</zorder>
   <zorder>label_17</zorder>
   <zorder>label_15</zorder>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
